### PR TITLE
[next] Use registerJSModule when possible

### PR DESCRIPTION
**Pull request by WebReflection**
## Description

As discussed after the TP, we decided to use best patterns for every interpreter so that polluting globals, as opposite of using a proper module to retrieve utilities, felt like the right thing to do.

This MR exposes in the **Python** world (both *MicroPython* and *Pyodide* interpreters) a way to retrieve the `xworker` utility from a worker code, or the `XWorker` constructor from the main thread.

## Changes

  * add a new `registerJSModule` *wrap* utility to help *core* define modules when possible
  * fallback to a globally shared reference in those interpreters not capable of registering a module (*Wasmoon* and *Ruby*)
  * change all integration tests to reflect this new way to reach either `XWorker` or `xworker` in *Python* interpreters
  * test manually all the things work as expected, even from a module fetched after a worker
  * took a chance to remove pyscript.js from git as that's an artifact and it should never bother the review

## Checklist

-   [x] All tests pass locally
-   [ ] I have updated `docs/changelog.md`
-   [x] I have created documentation for this(if applicable)

### CODE REVIEW

File: pyscript.core/docs/README.md
Comment (fpliger) [inline pyscript.core/docs/README.md:236]: As discussed today, this is a totally fine module name for now even if it may end up with another name or also available in the `pyscript` module itself (just want to repeat it out lout to make sure we are on the same page :) )

File: pyscript.core/docs/README.md
Comment (fpliger) [inline pyscript.core/docs/README.md:317]: I think I misunderstood this during our call today. I thought we'd be removing `XWorker` as a global reference and only keeping it as part of the module namespace. If we are keeping both, what users should use and why?

Comment (fpliger) [reply pyscript.core/docs/README.md:317]: Not that it's a big deal (especially right now), it's just a way to clarify and make sure we are pointing users at the "right way of doing things"

Comment (hoodmane) [reply pyscript.core/docs/README.md:317]: I read this as meaning that it depends on the runtime? So if the runtime doesn't have a convenient way to import xworker, it would just be added as a global?

File: pyscript.core/esm/interpreter/_utils.js
Comment (hoodmane) [inline pyscript.core/esm/interpreter/_utils.js:142]: @fpliger see here

**Review by fpliger — APPROVED** [review]
After leaving the comments, reading the changes, I think it answers my questions but will let you comment @WebReflection . I think this PR is good to merge either way and, if needed, we can amend the docs or recommendations (to "point users to the right way of doing things based on their Interpreter") in a follow up PR

File: pyscript.core/docs/README.md
Comment (hoodmane) [reply pyscript.core/docs/README.md:317]: @WebReflection I think it would be good to say "or as an import" or something like that since the wording "a `from xworker import xworker` is Python specific.

File: pyscript.core/esm/interpreter/_utils.js
Comment (fpliger) [reply pyscript.core/esm/interpreter/_utils.js:142]: Yeah, exactly. Saw that after I left my comments... but I think it's worth spelling out the intentions anyway just to make sure we are explicitly in agreement (without having to check the implementation :) )

File: pyscript.core/docs/README.md
Comment (WebReflection) [reply pyscript.core/docs/README.md:236]: I think `pyscript` module has a good place in ... 🥁🥁🥁 ... **pyscript** custom type :D ... this is core though, and we discussed this was a good name indeed for the core offer.

Comment (WebReflection) [reply pyscript.core/docs/README.md:236]: if this wasn't clear, **yes**, we can offer the same from **pyscript** custom type so there's no conflict of interest in here, I suppose

File: pyscript.core/docs/README.md
Comment (WebReflection) [reply pyscript.core/docs/README.md:317]: @fpliger what @hoodmane said ... it's **not possible** in *Wasmoon* or *Ruby* to register a *JS* module and consume it ... in those cases (code should speak for itself) stuff is added globally, as there's **no other way** to do so.

> I think it would be good to say "or as an import" or something like that since the wording "a from xworker import xworker is Python specific.

that's true but at the end of the day, we only have *Python* interpreters defined as stable, and these are the only one able to import modules ... so I've thought that having the wording 100% matching the code needed to make it work at this stage was convenient. I can change that though.

File: pyscript.core/esm/interpreter/_utils.js
Comment (WebReflection) [reply pyscript.core/esm/interpreter/_utils.js:142]: the intention is written in the code and, as mentioned in the previous comment, there's no other way to do the same in foreign, not fully capable, interpreters (unfortunately ... or due my not 100% understanding of their internals).

**Comment by WebReflection** [comment]
@fpliger thanks! I should've read everything before replying inline but it's good we're aligned. I am waiting for @hoodmane answer around the wording, as I think he has a point, and it takes little effort for me to phrase that better.

**Comment by WebReflection** [comment]
OK, my latest push should address @hoodmane hint/suggestion and I think we can move this forward and revisit, if needed, in a different MR.

**Comment by hoodmane** [comment]
Yeah new wording looks fine.

File: pyscript.core/docs/README.md
Comment (hoodmane) [reply pyscript.core/docs/README.md]: I will try to look into adding `registerJSModule` or something like it to wasmoon if I can get around to it.

Comment (fpliger) [reply pyscript.core/docs/README.md]: Yeah. What threw me off was the wording above. It reads as both are possibilities with **any** interpreter not that it's available as import for interpreters that support it and if the runtime doesn't have a convenient way to import xworker, it would just be added as a global

**Pull request merged**
